### PR TITLE
config: IPv6 v2 ResolvedAddresses no longer need square brackets.

### PR DESCRIPTION
**Pull request by htuch**
Use the "try IPv6, then try IPv4 if it fails" parse algorithm.

### CODE REVIEW

File: test/test_common/network_utility.cc [hunk 1]
```diff
@@ -88,6 +88,13 @@ const std::string getLoopbackAddressUrlString(const Address::IpVersion version)
...
+const std::string getLoopbackAddressString(const Address::IpVersion version) {
+  if (version == Address::IpVersion::v6) {
+    return std::string("::1");
+  }
+  return std::string("127.0.0.1");
+}
+
 const std::string getAnyAddressUrlString(const Address::IpVersion version) {
   if (version == Address::IpVersion::v6) {
     return std::string("[::]");
```
Comment (mattklein123) [inline test/test_common/network_utility.cc]: Do we still have other code that uses brackets? And I think we have tests that use brackets too? Can we just remove all bracket use? ntop_ip_loopback_address  vs. ip_loopback_address is kind of confusing to me.

Comment (htuch) [inline test/test_common/network_utility.cc]: Various integration tests use brackets to form URLs from substitutions, e.g. `tcp://{{ ip_loopback_address }}::{{ some_port }}`. We need brackets in this case, as this is the URI format.

Comment (mattklein123) [inline test/test_common/network_utility.cc]: Sorry I'm probably just being dense. When are brackets needed and when are they not? Is brackets something we made up or is that the way people always write v6 addresses?

Comment (mattklein123) [inline test/test_common/network_utility.cc]: Ah I see brackets are needed for ports. Got it. OK.

Comment (htuch) [inline test/test_common/network_utility.cc]: Yeah. From Wikipedia:

```
Colon (:) characters in IPv6 addresses may conflict with the established syntax of resource identifiers, such as URIs and URLs. The colon has traditionally been used to terminate the host path before a port number.[8] To alleviate this conflict, literal IPv6 addresses are enclosed in square brackets in such resource identifiers,```

**Pull request merged**